### PR TITLE
Update the website getting started docs

### LINKED ISSUE #2117: remove package import alias requirement
### Problem Statement

The requirement for import aliases makes symbol discovery hard (go-doc) and automatic import formatting (goimports) impossible.

### Proposed Solution

Since we are before the 1.0.0 release, I propose moving packages to their suggested aliases: e.g. `go.opentelemetry.io/otel/sdk/trace -> go.opentelemetry.io/otel/sdk/sdktrace`.

#### Alternatives

We could do nothing, pick a new globally unique name, or do this and also leave behind type aliases for compatibility.

#### Prior Art

The go standard library does this all the time: e.g. `net/http/httputil`.

#### goimports
For a given main package with no import statements:
```go
package main

func main() {
	e, err := stdouttrace.New(stdouttrace.WithPrettyPrint())
	if err != nil {
		log.Fatalf("configuring stdouttrace: %v", err)
	}
	p := sdktrace.NewTracerProvider()
	p.RegisterSpanProcessor(sdktrace.NewSimpleSpanProcessor(e))
	// ...
}
```
When `goimports` is run: (frequently a git hook or ide will do this for you)
```sh
[user@localhost ~]$ goimports -w .
```
Note that the toolchain is able to discover `stdouttrace`, but not `sdktrace`:
```go
package main

import (
	"log"

	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
)

func main() {
	e, err := stdouttrace.New(stdouttrace.WithPrettyPrint())
	if err != nil {
		log.Fatalf("configuring stdouttrace: %v", err)
	}
	p := sdktrace.NewTracerProvider()
	p.RegisterSpanProcessor(sdktrace.NewSimpleSpanProcessor(e))
	// ...
}
```
#### go-doc
Since the package names overlap, asking for certain symbols does not resolve as you would like:
```
[user@localhost ~]$ go doc trace.Tracer
package trace // import "go.opentelemetry.io/otel/sdk/trace"

func (p *TracerProvider) Tracer(name string, opts ...trace.TracerOption) trace.Tracer
    Tracer returns a Tracer with the given name and options. If a Tracer for the
    given name and options does not exist it is created, otherwise the existing
    Tracer is returned.

    If name is empty, DefaultTracerName is used instead.

    This method is safe to be called concurrently.
```
When what we wanted was:
```
[user@localhost ~]$ go doc otel/trace.Tracer
package trace // import "go.opentelemetry.io/otel/trace"

type Tracer interface {
	// Start creates a span and a context.Context containing the newly-created span.
	//
	// If the context.Context provided in `ctx` contains a Span then the newly-created
	// Span will be a child of that span, otherwise it will be a root span. This behavior
	// can be overridden by providing `WithNewRoot()` as a SpanOption, causing the
	// newly-created Span to be a root span even if `ctx` contains a Span.
	//
	// When creating a Span it is recommended to provide all known span attributes using
	// the `WithAttributes()` SpanOption as samplers will only have access to the
	// attributes provided when a Span is created.
	//
	// Any Span that is created MUST also be ended. This is the responsibility of the user.
	// Implementations of this API may leak memory or other resources if Spans are not ended.
	Start(ctx context.Context, spanName string, opts ...SpanStartOption) (context.Context, Span)
}
    Tracer is the creator of Spans.

    Warning: methods may be added to this interface in minor releases.
```

### PR DESCRIPTION
**Pull request by MrAlias**
Add a new example, fib, that contains an application for the computation of Fibonacci numbers. Use this example to update the website getting started documentation.

Resolves #2117

### CODE REVIEW

File: website_docs/getting-started.md [hunk 1]
```diff
@@ -3,216 +3,523 @@ title: "Getting Started"
...
 ```
 
-This creates a new console exporter with basic options - `WithPrettyPrint` formats the text nicely when its printed, so that it's easier for humans to read.
+Using the full-qualified package name, something that should be unique for Go packages, is the standard way to identify a `Tracer`. If your example package name differs, be sure to update the name you use here to match.
 
-## Creating a Tracer Provider
+Everything should be in place now to start tracing your application. But first, what is a trace? And, how exactly should you build them for you application?
 
-A trace is a type of telemetry that represents work being done by a service. In a distributed system, a trace can be thought of as a 'stack trace', showing the work being done by each service as well as the upstream and downstream calls that its making to other services.
+To back up a bit, a trace is a type of telemetry that represents work being done by a service. In a distributed system, a trace can be thought of as a 'stack trace', showing the work being done by each service as well as the upstream and downstream calls that its making to other services.
```
Comment (jmacd) [inline website_docs/getting-started.md:174]:
```suggestion
To back up a bit, a trace is a type of telemetry that represents work being done by a service. A trace is a record of the connection(s) between participants processing a transaction, often through client/server requests processing and other forms of communication.
```

File: website_docs/getting-started.md [hunk 2]
```diff
@@ -3,216 +3,523 @@ title: "Getting Started"
...
 
-OpenTelemetry requires a trace provider to be initialized in order to generate traces. A trace provider can have multiple span processors, which are components that allow for span data to be modified or exported after it's created.
+Each part of the work that a service performs is represented in the trace with a span. Those spans are not just an unordered collection. Like the call stack of our application, those spans are defined with relationships to one another. If that last part about span relationships doesn't make to much sense now, don't worry. The important thing to take away is each part of your code that does work should to be represented as a span. You will have a better understanding of these span relationships after you instrument your code, so let's get started.
```
Comment (jmacd) [inline website_docs/getting-started.md:176]:
```suggestion
Each part of the work that a service performs is represented in the trace by a span. The "root" span is the one that initiates the request, and except for the root span all spans have a parent relationship to another span in the same trace. The important thing to take away is each part of your code that does work should to be represented as a span. You will have a better understanding of these span relationships after you instrument your code, so let's get started.
```

**Review by jmacd — APPROVED** [review]

File: example/fib/app.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	defer span.End()
+
+	a.l.Print("What Fibonacci number would you like to know: ")
+
+	var n uint
+	_, err := fmt.Fscanf(a.r, "%d", &n)
+	if err != nil {
+		span.RecordError(err)
+		span.SetStatus(codes.Error, err.Error())
+	}
```
Comment (evantorrie) [inline example/fib/app.go:73]: Rather than continuing on here, shouldn't it just `return 0, err`

**Review by evantorrie — COMMENTED** [review]
Only one minor suggestion re: error handling

File: example/fib/main.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	)
+	defer func() {
+		if err := tp.Shutdown(context.Background()); err != nil {
+			l.Fatal(err)
+		}
+	}()
+	otel.SetTracerProvider(tp)
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt)
```
Comment (seh) [inline example/fib/main.go:84]: Consider using [the `signal.NotifyContext` function](https://pkg.go.dev/os/signal#NotifyContext) instead.

**Pull request merged**